web/packages/design/src/DataTable/useTable.ts

```diff
@@ -42,46 +42,71 @@ export default function useTable<T>({
-    pagination?: Pagination<T>;
-  }>(() => {
+  // Determine the initial sort
+  let initialSort: Sort<T> | undefined;
+  if (!customSort) {
+    const { initialSort: initialSortProp } = props;
     // Finds the first sortable column to use for the initial sorting
     let col: TableColumn<T> | undefined;
-    if (!customSort) {
-      const { initialSort } = props;
-      if (initialSort) {
-        col = initialSort.altSortKey
-          ? columns.find(col => col.altSortKey === initialSort.altSortKey)
-          : columns.find(col => col.key === initialSort.key);
-      } else {
-        col = columns.find(column => column.isSortable);
-      }
+    if (initialSortProp) {
+      col = initialSortProp.altSortKey
+        ? columns.find(col => col.altSortKey === initialSortProp.altSortKey)
+        : columns.find(col => col.key === initialSortProp.key);
+    } else {
+      col = columns.find(column => column.isSortable);
     }
+    if (col) {
+      initialSort = {
+        key: (col.altSortKey || col.key) as keyof T,
+        onSort: col.onSort,
+        dir: initialSortProp?.dir || 'ASC',
+      };
+    }
+  }
 
-    return {
-      data: [],
-      searchValue: clientSearch?.initialSearchValue || '',
-      sort: col
-        ? {
-            key: (col.altSortKey || col.key) as keyof T,
-            onSort: col.onSort,
-            dir: props.initialSort?.dir || 'ASC',
-          }
-        : undefined,
-      pagination: pagination
-        ? {
-            paginatedData: paginateData([], pagination.pageSize),
-            currentPage: 0,
-            pagerPosition: pagination.pagerPosition,
-            pageSize: pagination.pageSize || 15,
-            CustomTable: pagination.CustomTable,
-          }
-        : undefined,
+  // Compute the initial data
+  const initialSearchValue = clientSearch?.initialSearchValue || '';
+  let initialData: T[];
+  if (serversideProps || disableFilter || !data?.length) {
+    initialData = data || [];
+  } else {
+    initialData = sortAndFilter(
+      data,
+      initialSearchValue,
+      initialSort,
+      searchableProps ||
+        (columns
+          .filter(column => column.key)
+          .map(column => column.key) as (keyof T & string)[]),
+      searchAndFilterCb,
+      showFirst
+    );
+  }
+
+  // Compute initial pagination if applicable
+  let initialPagination: Pagination<T> | undefined;
+  if (pagination) {
+    const pages = paginateData(initialData, pagination.pageSize);
+    initialPagination = {
+      paginatedData: pages,
+      currentPage: 0,
+      pagerPosition: pagination.pagerPosition,
+      pageSize: pagination.pageSize || 15,
+      CustomTable: pagination.CustomTable,
     };
-  });
+  }
+
+  const [state, setState] = useState<{
+    data: T[];
+    searchValue: string;
+    sort?: Sort<T>;
+    pagination?: Pagination<T>;
+  }>(() => ({
+    data: initialData,
+    searchValue: initialSearchValue,
+    sort: initialSort,
+    pagination: initialPagination,
+  }));
```

Comment:
it looks like these `initialXXX` is only used initially, so should all the computing of these initial value be inside here to do it only once like it did before? otherwise for after initial re-renders, these initial values get re-computed, but it's not being used

Reply:
True, didn't catch that. That'd be better